src/components/waitList/WaitListForm.tsx

```diff
@@ -0,0 +1,238 @@
+
+interface FormData {
+  full_name: string;
+  email: string;
+}
+
+interface FormErrors {
+  full_name?: string;
+  email?: string;
+  server?: string;
+}
+
+const WaitlistForm: React.FC = () => {
+  const [formData, setFormData] = useState<FormData>({
+    full_name: "",
+    email: "",
+  });
+  const [errors, setErrors] = useState<FormErrors>({});
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const timeoutReference = useRef<NodeJS.Timeout | null>(null);
+
+  type FormErrorKey = keyof FormErrors;
+
+  useEffect(() => {
+    // Clear the timeout if the component unmounts
+    return () => {
+      if (timeoutReference.current) {
+        clearTimeout(timeoutReference.current);
+      }
+    };
+  }, []);
+
+  const validate = (): FormErrors => {
+    const newErrors: FormErrors = {};
+
+    const validationRules: {
+      check: boolean;
+      field: FormErrorKey;
+      message: string;
+    }[] = [
+      {
+        check: formData.full_name.trim() === "",
+        field: "full_name",
+        message: "Your full name is required here",
+      },
+      {
+        check: formData.email.trim() === "",
+        field: "email",
+        message: "Please enter a valid email address",
+      },
+      {
+        check:
+          formData.email.trim() !== "" && !/\S+@\S+\.\S+/.test(formData.email),
+        field: "email",
+        message: "Please enter a valid email address",
+      },
+    ];
+
+    for (const rule of validationRules) {
+      if (rule.check) {
+        newErrors[rule.field] = rule.message;
+      }
+    }
+
+    return newErrors;
+  };
+
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newErrors = validate();
+    if (Object.keys(newErrors).length > 0) {
+      setErrors(newErrors);
+    } else {
+      setErrors({});
+      try {
+        const response = await fetch(
+          process.env.NEXT_PUBLIC_API_URL as string,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              full_name: formData.full_name,
+              email: formData.email,
+            }),
+          },
+        );
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          if (errorData.message.message === "Email already registered") {
+            setErrors({ server: "Email already registered" });
+          } else {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+          }
+          return;
+        }
+
+        setFormData({ full_name: "", email: "" });
+        setIsSubmitted(true);
+
+        // Set a timeout to reset the form after 5 seconds
+        timeoutReference.current = setTimeout(() => {
+          setIsSubmitted(false);
+        }, 6000);
+      } catch {
+        // console.error("There was a problem with the fetch operation:", error);
```

Comment:
Remove unnecessary comments from the implementation as this will make it more professional

Reply:
okay done